File: Rivington.IG.Web/Rivington.IG.API/src/app/modules/shared/models/ordermanagement/inspection-order/checklist/property/building-concern.ts
import { InspectionOrderProperty } from "../property";
import { InspectionOrderPropertyBuildingConcernElectricalConcernDetails } from "./building-concern/InspectionOrderPropertyBuildingConcernElectricalConcernDetails";
import { InspectionOrderPropertyBuildingConcernExteriorBuildingConcernDetails } from "./building-concern/InspectionOrderPropertyBuildingConcernExteriorBuildingConcernDetails";
import { InspectionOrderPropertyBuildingConcernOtherStructureConcernDetails } from "./building-concern/InspectionOrderPropertyBuildingConcernOtherStructureConcernDetails";
import { InspectionOrderPropertyBuildingConcernPlumbingConcernDetails } from "./building-concern/InspectionOrderPropertyBuildingConcernPlumbingConcernDetails";
import { InspectionOrderPropertyBuildingConcernRoofConcernDetails } from "./building-concern/InspectionOrderPropertyBuildingConcernRoofConcernDetails";
import { InspectionOrderPropertyBuildingConcernSurroundingAreaConcernDetails } from "./building-concern/InspectionOrderPropertyBuildingConcernSurroundingAreaConcernDetails";

export class InspectionOrderPropertyBuildingConcern {
    /*[Parent]*/
    inspectionOrderProperty?: InspectionOrderProperty;
    /*[Key]*/
    id?: string;
    exteriorBuildingConcernDetails?: InspectionOrderPropertyBuildingConcernExteriorBuildingConcernDetails[];
    roofConcernDetails?: InspectionOrderPropertyBuildingConcernRoofConcernDetails[];
    electricalConcernDetails?: InspectionOrderPropertyBuildingConcernElectricalConcernDetails[];
    plumbingConcernDetails?: InspectionOrderPropertyBuildingConcernPlumbingConcernDetails[];
    otherStructureConcernDetails?: InspectionOrderPropertyBuildingConcernOtherStructureConcernDetails[];
    surroundingAreaConcernDetails?: InspectionOrderPropertyBuildingConcernSurroundingAreaConcernDetails[];
    public constructor(init?:Partial<InspectionOrderPropertyBuildingConcern>) {
        Object.assign(this, init);
    }
  }

File: Rivington.IG.Web/Rivington.IG.Domain.Models/OrderManagement/InspectionOrderProperty/BuildingConcern.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace Rivington.IG.Domain.Models.OrderManagement
{
    public class InspectionOrderPropertyBuildingConcern
    {
        [Parent]
        public InspectionOrderProperty InspectionOrderProperty { get; set; }
        [Key]
        public Guid Id { get; set; }

        public ICollection<InspectionOrderPropertyBuildingConcernExteriorBuildingConcernDetails> ExteriorBuildingConcernDetails { get; set; }
        public ICollection<InspectionOrderPropertyBuildingConcernRoofConcernDetails> RoofConcernDetails { get; set; }
        public ICollection<InspectionOrderPropertyBuildingConcernElectricalConcernDetails> ElectricalConcernDetails { get; set; }
        public ICollection<InspectionOrderPropertyBuildingConcernPlumbingConcernDetails> PlumbingConcernDetails { get; set; }
        public ICollection<InspectionOrderPropertyBuildingConcernOtherStructureConcernDetails> OtherStructureConcernDetails { get; set; }
        public ICollection<InspectionOrderPropertyBuildingConcernSurroundingAreaConcernDetails> SurroundingAreaConcernDetails { get; set; }

        public static void BuildModel(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<InspectionOrderProperty>()
                .HasOne(a => a.BuildingConcern).WithOne(b => b.InspectionOrderProperty)
                .HasForeignKey<InspectionOrderPropertyBuildingConcern>()
                .HasConstraintName($"FK_{nameof(InspectionOrderProperty)}_{nameof(InspectionOrderPropertyBuildingConcern)}")
                .OnDelete(DeleteBehavior.Cascade);

            InspectionOrderPropertyBuildingConcernExteriorBuildingConcernDetails.BuildModel(modelBuilder);

            InspectionOrderPropertyBuildingConcernRoofConcernDetails.BuildModel(modelBuilder);

            InspectionOrderPropertyBuildingConcernElectricalConcernDetails.BuildModel(modelBuilder);

            InspectionOrderPropertyBuildingConcernPlumbingConcernDetails.BuildModel(modelBuilder);

            InspectionOrderPropertyBuildingConcernOtherStructureConcernDetails.BuildModel(modelBuilder);

            InspectionOrderPropertyBuildingConcernSurroundingAreaConcernDetails.BuildModel(modelBuilder);
        }
    }
}